### PR TITLE
BENCH 281 - Create Order Controller unit tests

### PR DESCRIPTION
**Pull request by hazel-davis**

### CODE REVIEW

**Comment by milkshakeuk** [comment]
branch needs rebasing I think and conflicts resolving.

File: tests/Answer.King.Api.UnitTests/Controllers/CategoriesControllerTests.cs [hunk 1]
```diff
@@ -3,15 +3,13 @@
...
 using Answer.King.Test.Common.CustomAsserts;
-using Answer.King.Test.Common.CustomTraits;
 using Microsoft.AspNetCore.Mvc;
 using NSubstitute;
 using NSubstitute.ExceptionExtensions;
 using Xunit;
 
 namespace Answer.King.Api.UnitTests.Controllers;
 
-[TestCategory(TestType.Unit)]
```
Comment (milkshakeuk) [inline tests/Answer.King.Api.UnitTests/Controllers/CategoriesControllerTests.cs]: Any reason you have removed this?

File: tests/Answer.King.Api.UnitTests/Controllers/OrdersControllerTests.cs [hunk 2]
```diff
@@ -1,6 +1,218 @@
...
+using NSubstitute.ReceivedExtensions;
+using Xunit;
+using InputOrder = Answer.King.Api.RequestModels.Order;
+using LineItem = Answer.King.Api.RequestModels.LineItem;
+using OutputOrder = Answer.King.Domain.Orders.Order;
+
+namespace Answer.King.Api.UnitTests.Controllers;
 
-// Todo: Write Order Controller unit tests
 public class OrdersControllerTests
```
Comment (milkshakeuk) [inline tests/Answer.King.Api.UnitTests/Controllers/OrdersControllerTests.cs:16]: No Test Category?

File: tests/Answer.King.Api.UnitTests/Controllers/OrdersControllerTests.cs [hunk 3]
```diff
@@ -1,6 +1,218 @@
...
+    #endregion GetAll
+
+    #region GetOne
+
+    [Fact]
+    public void GetOne_Method_HasCorrectVerbAttributeAndPath()
+    {
+        // Assert
+        AssertController.MethodHasVerb<OrdersController, HttpGetAttribute>(
+            nameof(OrdersController.GetOne), "{id:long}");
```
Comment (milkshakeuk) [inline tests/Answer.King.Api.UnitTests/Controllers/OrdersControllerTests.cs]: you should remove the type hint here as it changes the behaviour of the endpoints.
https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/routing?view=aspnetcore-7.0#attribute-routing-with-http-verb-attributes

File: tests/Answer.King.Api.UnitTests/Controllers/OrdersControllerTests.cs [hunk 4]
```diff
@@ -1,6 +1,218 @@
...
+    [Fact]
+    public void GetAll_Method_HasCorrectVerbAttribute()
+    {
+        // Assert
+        AssertController.MethodHasVerb<OrdersController, HttpGetAttribute>(
+            nameof(OrdersController.GetAll));
+    }
+
+    [Fact]
+    public async void GetAll_ValidRequest_ReturnsOkObjectResult()
```
Comment (bethcryer) [inline tests/Answer.King.Api.UnitTests/Controllers/OrdersControllerTests.cs]: Any `async void` test methods need to be changed to `async Task`

**Review by bethcryer — CHANGES_REQUESTED** [review]

**Review by milkshakeuk — CHANGES_REQUESTED** [review]

**Review by bethcryer — APPROVED** [review]

**Review by milkshakeuk — APPROVED** [review]

**Pull request merged**